v2/tools/generator/internal/codegen/pipeline/resource_registration_file.go

```diff
@@ -378,6 +391,75 @@ func createGetKnownStorageTypesFunc(
 	return f.DefineFunc()
 }
 
+func (r *ResourceRegistrationFile) createGetResourceExtensions(context *astmodel.CodeGenerationContext) (dst.Decl, error) {
+
+	return createGetResourceExtensionsImpl(
+		context,
+		r.resourceExtensions,
+		"getResourceExtensions",
+		"returns a list of Resource Extension types")
+
+}
+
+func createGetResourceExtensionsImpl(codeGenerationContext *astmodel.CodeGenerationContext, resources []astmodel.TypeName, funcName string, funcComment string) (dst.Decl, error) {
+	// Sort the resources for a deterministic file layout
+	sort.Slice(resources, orderByImportedTypeName(codeGenerationContext, resources))
+	packageRef, err := codeGenerationContext.GetImportedPackageName(astmodel.GenRuntimeReference)
+	if err != nil {
+		return nil, err
+	}
+	resultIdent := dst.NewIdent("result")
+	resultVar := astbuilder.LocalVariableDeclaration(
+		resultIdent.String(),
+		&dst.ArrayType{
+			Elt: &dst.SelectorExpr{
+				X:   dst.NewIdent(packageRef),
+				Sel: dst.NewIdent("ResourceExtension"),
+			},
+		},
+		"")
+
+	var resourceAppendStatements []dst.Stmt
+	for _, typeName := range resources {
+		appendStmt := astbuilder.AppendSlice(
+			resultIdent,
+			astbuilder.AddrOf(astbuilder.NewCompositeLiteralBuilder(typeName.AsType(codeGenerationContext)).Build()),
+		)
+		resourceAppendStatements = append(resourceAppendStatements, appendStmt)
+	}
+
+	returnStmt := &dst.ReturnStmt{
+		Results: []dst.Expr{
+			resultIdent,
+		},
+	}
+
+	var body []dst.Stmt
+	body = append(body, resultVar)
+	body = append(body, resourceAppendStatements...)
+	body = append(body, returnStmt)
+
+	f := &astbuilder.FuncDetails{
+		Name:   funcName,
+		Body:   body,
+		Params: []*dst.Field{},
+		Returns: []*dst.Field{
+			{
+				Type: &dst.ArrayType{
+					Elt: &dst.SelectorExpr{
+						X:   dst.NewIdent(packageRef),
+						Sel: dst.NewIdent("ResourceExtension"),
+					},
+				},
+			},
+		},
+	}
```

Comment:
Lean in to the helper methods, they reduce boilerplate. 
```suggestion
	f := &astbuilder.FuncDetails{
		Name:   funcName,
		Body:   body,
	}
	f.AddReturn(astbuilder.QualifiedTypeName(packageRef, "ResourceExtension"))
```